### PR TITLE
Add concise and detailed logs for shunt compensators activation

### PR DESCRIPTION
**Pull request by FranckLecuyer**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

No logs are generated for shunt compensators, when the delta between the discretized reactive value and the optimal reactive value given by the optimizer is too big, and thus can lead to further loadflow divergence problem.

**What is the new behavior (if this is a feature change)?**
Concise and detailed logs are now generated for shunt compensators activation, when the delta between the discretized reactive value and the optimal reactive value is over a configured threshold.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

**Review by ayolab — APPROVED** [review]

**Review by So-Fras — CHANGES_REQUESTED** [review]

File: open-reac/src/test/java/com/powsybl/openreac/OpenReacRunnerTest.java [hunk 1]
```diff
@@ -259,39 +262,39 @@ void testRunAsync() throws IOException {
         }
     }
 
-    private void testAllModifAndLoadFlow(Network network, String subFolder, OpenReacParameters parameters) throws IOException {
-        runAndApplyAllModifications(network, subFolder, parameters, true);
+    private void testAllModifAndLoadFlow(Network network, String subFolder, OpenReacParameters parameters, ReportNode reportNode) throws IOException {
+        runAndApplyAllModifications(network, subFolder, parameters, true, reportNode);
         LoadFlowResult loadFlowResult = LoadFlow.run(network);
-        assertTrue(loadFlowResult.isOk());
+        assertTrue(loadFlowResult.isFullyConverged() || loadFlowResult.isPartiallyConverged());
```
Comment (So-Fras) [inline open-reac/src/test/java/com/powsybl/openreac/OpenReacRunnerTest.java:268]: I saw that `isOk()` is deprecated. But is that function equivalent to `isFullyConverged() || isPartiallyConverged()`?

File: open-reac/src/main/java/com/powsybl/openreac/parameters/output/network/NetworkModifications.java [hunk 2]
```diff
@@ -48,6 +48,10 @@ public List<ShuntCompensatorModification> getShuntModifications() {
         return shuntsOutput.getModifications();
     }
 
+    public List<ShuntCompensatorNetworkOutput.ShuntWithDeltaDiscreteOptimalOverThrehold> getShuntsWithDeltaDiscreteOptimalOverThrehold() {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply open-reac/src/main/java/com/powsybl/openreac/parameters/output/network/NetworkModifications.java]: Fixed

File: open-reac/src/main/java/com/powsybl/openreac/parameters/output/network/ShuntCompensatorNetworkOutput.java [hunk 3]
```diff
@@ -21,9 +24,14 @@ public class ShuntCompensatorNetworkOutput extends AbstractNetworkOutput<ShuntCo
     private static final int ID_COLUMN_INDEX = 1;
     private static final int B_COLUMN_INDEX = 3;
     private static final int BUS_COLUMN_INDEX = 2;
+    private final List<ShuntWithDeltaDiscreteOptimalOverThrehold> shuntWithDeltaDiscreteOptimalOverThreholds = new ArrayList<>();
+    private final double shuntCompensatorActivationAlertThreshold;
+
+    public record ShuntWithDeltaDiscreteOptimalOverThrehold(String id, int maximumSectionCount, double discretizedReactiveValue, double optimalReactiveValue) { }
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply open-reac/src/main/java/com/powsybl/openreac/parameters/output/network/ShuntCompensatorNetworkOutput.java]: Fixed

File: open-reac/src/main/java/com/powsybl/openreac/parameters/output/network/ShuntCompensatorNetworkOutput.java [hunk 4]
```diff
@@ -64,6 +72,15 @@ private int findSectionCount(ShuntCompensator sc, double b) {
...
         }
+        double optimalReactiveValue = Math.abs(b * Math.pow(sc.getTerminal().getVoltageLevel().getNominalV(), 2));
+        double discretizedReactiveValue = Math.abs(sc.getB(sectionCount) * Math.pow(sc.getTerminal().getVoltageLevel().getNominalV(), 2));
+        if (Math.abs(discretizedReactiveValue - optimalReactiveValue) > shuntCompensatorActivationAlertThreshold) {
+            shuntWithDeltaDiscreteOptimalOverThreholds.add(new ShuntWithDeltaDiscreteOptimalOverThrehold(sc.getId(), sc.getMaximumSectionCount(), discretizedReactiveValue, optimalReactiveValue));
+        }
         return sectionCount;
     }
+
+    public List<ShuntWithDeltaDiscreteOptimalOverThrehold> getShuntsWithDeltaDiscreteOptimalOverThreholds() {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply open-reac/src/main/java/com/powsybl/openreac/parameters/output/network/ShuntCompensatorNetworkOutput.java]: Fixed

File: open-reac/src/main/java/com/powsybl/openreac/parameters/output/network/ShuntCompensatorNetworkOutput.java [hunk 5]
```diff
@@ -64,6 +72,15 @@ private int findSectionCount(ShuntCompensator sc, double b) {
...
+        return shuntWithDeltaDiscreteOptimalOverThreholds;
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply open-reac/src/main/java/com/powsybl/openreac/parameters/output/network/ShuntCompensatorNetworkOutput.java]: Fixed

File: open-reac/src/test/java/com/powsybl/openreac/OpenReacRunnerTest.java (hunk 1 above)
Comment (FranckLecuyer) [reply open-reac/src/test/java/com/powsybl/openreac/OpenReacRunnerTest.java]: I have changed this back to call isOk(), even if it is deprecated (isOk() is still called in powsybl-core ...)
Is this acceptable ?

Comment (FranckLecuyer) [reply open-reac/src/test/java/com/powsybl/openreac/OpenReacRunnerTest.java]: Replaced by a call to isFullyConverged (same was previously done in powsybl-open-loadflow)

**Review by So-Fras — APPROVED** [review]

**Pull request merged**